packages/blade/src/components/Input/BaseInput/__tests__/BaseInput.web.test.tsx

```diff
@@ -0,0 +1,177 @@
+    const errorText = getByText('Error');
+
+    expect(errorText).toBeTruthy();
+    expect(input).toHaveAccessibleDescription('Error');
+    expect(input).toBeInvalid();
+  });
+
+  it('should render with icons', () => {
+    const { container } = renderWithTheme(
+      <BaseInput
+        label="Enter name"
+        placeholder="First Last"
+        id="name"
+        leadingIcon={EyeIcon}
+        trailingIcon={CloseIcon}
+      />,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should be focussed when autoFocus flag is passed', () => {
+    const label = 'Enter name';
+    // eslint-disable-next-line jsx-a11y/no-autofocus
+    const { getByLabelText } = renderWithTheme(<BaseInput label={label} id="name" autoFocus />);
+
+    const input = getByLabelText(label);
+    expect(input).toHaveFocus();
+  });
+
+  it('should be disabled when isDisabled flag is passed', () => {
+    const label = 'Enter name';
+    const { getByLabelText } = renderWithTheme(<BaseInput label={label} id="name" isDisabled />);
+
+    const input = getByLabelText(label);
+    expect(input).toBeDisabled();
+  });
+
+  it('should handle onChange', async () => {
+    const label = 'Enter name';
+    const onChange = jest.fn();
+    const user = userEvent.setup();
+    const userName = 'Divyanshu';
+
+    const { getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" name="name" onChange={onChange} />,
+    );
+
+    const input = getByLabelText(label);
+    await user.type(input, userName);
+
+    // should be called for each keystroke
+    expect(onChange).toHaveBeenCalledTimes(userName.length);
+    expect(onChange).toHaveBeenLastCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should handle onBlur', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const userName = 'Divyanshu';
+    const onBlur = jest.fn();
+
+    renderWithTheme(
+      <BaseInput
+        label={label}
+        id="name"
+        name="name"
+        defaultValue={userName}
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus
+        onBlur={onBlur}
+      />,
+    );
+
+    // shifts user focus and therefore blurs the focussed input
+    await user.tab();
+    expect(onBlur).toHaveBeenCalledTimes(1);
+    expect(onBlur).toHaveBeenCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should set value as an uncontrolled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu Maithani';
+
+    const { getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" defaultValue={valueInitial} />,
+    );
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    await user.type(input, ' Maithani');
+    expect(input).toHaveValue(valueFinal);
+  });
+
+  it('should set value as a controlled input', () => {
```

Comment:
Can we modify this test a bit? 
It's better to test controlled component as how the user might use it in real-world. 

```ts

  it('should set value as a controlled input', async () => {
    const user = userEvent.setup();
    const label = 'Enter name';
    const valueInitial = 'Divyanshu';
    const valueFinal = 'Divyanshu Maithani';

    const Example = (): React.ReactElement => {
      const [value, setValue] = React.useState(valueInitial);

      return (
        <>
          <BaseInput
            label={label}
            id="name"
            value={value}
            onChange={({ value }) => setValue(value!)}
          />
          <p data-testid="input-value">{value}</p>
        </>
      );
    };

    const { getByLabelText, getByTestId } = renderWithTheme(<Example />);

    const input = getByLabelText(label);
    expect(input).toHaveValue(valueInitial);
    await user.type(input, ' Maithani');

    expect(getByTestId('input-value')).toHaveTextContent(valueFinal);
  });
```

Reply:
We are testing if the value is actually controlled or not. 

If BaseInput for some reason stops changing the onChange value or doesn't take `value` prop as initial or we change some logic inside the BaseInput we will know since `input-value` won't be as we expect then.

Reply:
Done in https://github.com/razorpay/blade/pull/695/commits/8be6b8367acaeaf5f157a5ba38eb97ff0c75f785